### PR TITLE
feat: add new rule rrd ~ huge files with tests (#201)

### LINKED ISSUE #201: add rule (rrd) huge files
script + template + css together
warning on 300 lines and error on 500 lines

### PR DESCRIPTION
**Pull request by David-Pena**
### Summary
Add new rule huge files for rrd ruleset with tests

### Description
- Add rule file plus tests
- Add rule docs  

### Related Issues
Fixes #201 

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation update

### CODE REVIEW

File: docs/rules/rrd/huge-files.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Huge Files
+
+Checks if the combined length of script, template, and CSS in a Vue component is too long. The default threshold for a warning is 300 lines, and for an error is 500 lines.
```
Comment (rrd108) [inline docs/rules/rrd/huge-files.md:3]: In other rules error is the double of the warning. Can we do it here also? Or did you have some reason for these numbers?

File: src/rulesCheck.ts [hunk 2]
```diff
@@ -58,6 +58,7 @@ export const checkRules = (descriptor: SFCDescriptor, filePath: string, apply: s
     functionSize: () => checkFunctionSize(script, filePath, limits.maxFunctionSize),
     htmlImageElements: () => getIsNuxt() && checkHtmlImageElements(descriptor.template, filePath),
     htmlLink: () => isVueFile && checkHtmlLink(descriptor.template, filePath),
+    hugeFiles: () => isVueFile && checkHugeFiles(descriptor, filePath),
```
Comment (rrd108) [inline src/rulesCheck.ts:61]: Can we do it to all files, composables, api routes, etc?

File: src/rules/rrd/hugeFiles.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import type { SFCDescriptor } from '@vue/compiler-sfc'
+import type { FileCheckResult, Offense } from '../../types'
+
+const results: FileCheckResult[] = []
+
+const WARNING_THRESHOLD = 300
```
Comment (rrd108) [inline src/rules/rrd/hugeFiles.ts:6]: Add this to the override issue please

**Review by rrd108 — CHANGES_REQUESTED** [review]

File: docs/rules/rrd/huge-files.md (hunk 1 above)
Comment (David-Pena) [reply docs/rules/rrd/huge-files.md:3]: Oh the numbers were provided by you in the issue description 👀 but you are right, let's keep using the 2 * WARNING TRESHOLD

File: src/rulesCheck.ts (hunk 2 above)
Comment (David-Pena) [reply src/rulesCheck.ts:61]: Yes we can, I'll just have to add a new parameter so the check function knows if it's a Vue or js/ts file to modify the total lines calc

**Comment by rrd108** [comment]
It is fixed in the next commit

**Review by rrd108 — CHANGES_REQUESTED** [review]
Request doc change

File: docs/rules/rrd/huge-files.md (hunk 1 above)
Comment (David-Pena) [reply docs/rules/rrd/huge-files.md]: fixed in last commit

**Pull request merged**